paimon-core/src/main/java/org/apache/paimon/table/ExpireSnapshotsImpl.java

```diff
@@ -186,17 +186,6 @@ public int expireUntil(long earliestId, long endExclusiveId) {
             snapshotDeletion.cleanUnusedDataFiles(snapshot, skipper);
         }
 
-        // delete changelog files
-        for (long id = beginInclusiveId; id < endExclusiveId; id++) {
-            if (LOG.isDebugEnabled()) {
-                LOG.debug("Ready to delete changelog files from snapshot #" + id);
-            }
-            Snapshot snapshot = snapshotManager.snapshot(id);
-            if (snapshot.changelogManifestList() != null) {
```

Comment:
If possible, we should try to ensure that the delta file of the Append snapshot can be stream read when there is no changelog file present.

Reply:
Agree, will give a try.